src/main/scala/rocket/CSR.scala

```diff
@@ -475,7 +475,15 @@ class CSRFile(perfEventSets: EventSets = new EventSets(Seq()))(implicit p: Param
   val trapToDebug = Bool(usingDebug) && (reg_singleStepped || causeIsDebugInt || causeIsDebugTrigger || causeIsDebugBreak || reg_debug)
   val debugTVec = Mux(reg_debug, Mux(insn_break, UInt(0x800), UInt(0x808)), UInt(0x800))
   val delegate = Bool(usingVM) && reg_mstatus.prv <= PRV.S && Mux(cause(xLen-1), reg_mideleg(cause_lsbs), reg_medeleg(cause_lsbs))
-  val tvec = Mux(trapToDebug, debugTVec, Mux(delegate, reg_stvec.sextTo(vaddrBitsExtended), reg_mtvec))
+  val notDebugTVec = {
+    val base = Mux(delegate, reg_stvec.sextTo(vaddrBitsExtended), reg_mtvec)
+    val baseAlign = 2
+    val interruptAlign = log2Ceil(new MIP().getWidth)
+    val interruptOffset = cause(log2Ceil(new MIP().getWidth)-1, 0) << baseAlign
```

Comment:
Sure. But these are the people who complained about the sequence: "load, shift, add, jump". You only reduced it by *3* instructions when you could have saved *4*. Drams!